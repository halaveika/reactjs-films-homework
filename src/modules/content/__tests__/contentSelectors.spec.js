import {
  getSearchMovieResult, getMoviesGenre, getVideo,
  getSearchList, getDetailsPageSelector, getInitialisation,
} from '../contentSelectors';

const mockState = {
  content: {
    results: [{
      id: 2412412,
      title: 'Fight club',
      genre_ids: [13, 20, 21],
      vote_average: 8,
      poster: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    },
    {
      id: 41412,
      title: 'Sopranos',
      genre_ids: [13, 18],
      vote_average_ids: 10,
      poster: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    }],
    page: 1,
    total_pages: 12,
    total_results: 55,
    genres_array: [{
      id: 13,
      name: 'Adventure',
    }, {
      id: 18,
      name: 'Drama',
    }, {
      id: 20,
      name: 'Family',
    }, {
      id: 21,
      name: 'Fantasy',
    }],
    video_url: 'test_url',
    details: {
      id: 2412412,
      title: 'Fight club',
      runtime: 120,
      vote_average_ids: 10,
      poster: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing',
    },
    isInitialisated: false,
  },
};

const mockStateEmpty = {
  content: {
    page: 0,
    results: [],
    total_pages: 0,
    total_results: 0,
    genres_array: [],
    video_url: '',
    details: {},
  },
};

describe('contentSelectors testing', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('getSearchMovieResult should return results from state', () => {
    expect(getSearchMovieResult(mockState)).toMatchSnapshot();
  });

  test('getInitialisation should return results from state', () => {
    expect(getInitialisation(mockState)).toMatchSnapshot();
  });

  test('getMoviesGenre should return genres_array from state', () => {
    expect(getMoviesGenre(mockState)).toMatchSnapshot();
  });

  test('getVideo should return video_url from state', () => {
    expect(getVideo(mockState)).toMatchSnapshot();
  });

  test('getSearchList should return getSearchList props', () => {
    expect(getSearchList(mockState)).toMatchSnapshot();
  });

  test('getSearchList should correct handl empty result of search', () => {
    expect(getSearchList(mockStateEmpty)).toMatchSnapshot();
  });

  test('getDetailsPageSelector should return getDetailsPageSelector props', () => {
    expect(getDetailsPageSelector(mockState)).toMatchSnapshot();
  });
});
