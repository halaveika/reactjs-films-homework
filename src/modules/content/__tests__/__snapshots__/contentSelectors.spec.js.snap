// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentSelectors testing getDetailsPageSelector should return getDetailsPageSelector props 1`] = `
Object {
  "backdrop": undefined,
  "genres": Array [
    undefined,
    undefined,
    undefined,
  ],
  "id": 2412412,
  "overview": "Lorem Ipsum is simply dummy text of the printing",
  "runtime": 120,
  "title": "Fight club",
  "video": "test_url",
  "vote_average": 10,
}
`;

exports[`contentSelectors testing getDetailsSelector should return details from state 1`] = `
Object {
  "backdrop": "image_path",
  "genres": Array [
    "Drama",
    "Adventure",
    "Fantasy",
  ],
  "id": 2412412,
  "overview": "Lorem Ipsum is simply dummy text of the printing",
  "runtime": 120,
  "title": "Fight club",
  "vote_average": 10,
}
`;

exports[`contentSelectors testing getFilter should return filter from state 1`] = `"TRENDING"`;

exports[`contentSelectors testing getLoadingSelector should return loading status from state 1`] = `true`;

exports[`contentSelectors testing getMoviesGenre should return genres_array from state 1`] = `
Array [
  Object {
    "id": 13,
    "name": "Adventure",
  },
  Object {
    "id": 18,
    "name": "Drama",
  },
  Object {
    "id": 20,
    "name": "Family",
  },
  Object {
    "id": 21,
    "name": "Fantasy",
  },
]
`;

exports[`contentSelectors testing getPageSelector should return page  number from state 1`] = `1`;

exports[`contentSelectors testing getPageSizeSelector should return page size from state 1`] = `20`;

exports[`contentSelectors testing getSearchList should correct handl empty result of search 1`] = `Array []`;

exports[`contentSelectors testing getSearchList should return getSearchList props 1`] = `
Array [
  Object {
    "genres": Array [
      "Adventure",
      "Family",
      "Fantasy",
    ],
    "id": 2412412,
    "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry",
    "poster": "image_path",
    "title": "Fight club",
    "vote_average": 8,
  },
  Object {
    "genres": Array [
      "Adventure",
      "Drama",
    ],
    "id": 41412,
    "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry",
    "poster": "image_path",
    "title": "Sopranos",
    "vote_average": 10,
  },
]
`;

exports[`contentSelectors testing getSearchMovieResult should return results from state 1`] = `
Array [
  Object {
    "genre_ids": Array [
      13,
      20,
      21,
    ],
    "id": 2412412,
    "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry",
    "poster_path": "image_path",
    "title": "Fight club",
    "vote_average": 8,
  },
  Object {
    "genre_ids": Array [
      13,
      18,
    ],
    "id": 41412,
    "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry",
    "poster_path": "image_path",
    "title": "Sopranos",
    "vote_average": 10,
  },
]
`;

exports[`contentSelectors testing getSearchValueSelector should return search value from state 1`] = `""`;

exports[`contentSelectors testing getTotalPageSelector should return total page  number from state 1`] = `12`;

exports[`contentSelectors testing getVideo should return video_url from state 1`] = `"test_url"`;
